autoload/ale/handlers/deadnix.vim

```diff
@@ -0,0 +1,23 @@
+function! ale#handlers#deadnix#Handle(buffer, lines) abort
+    let l:output = []
+
+    for l:line in a:lines
+        try
+            let l:file = json_decode(l:line)
```

Comment:
Use our `ale#util#FuzzyJSONDecode` function. It tends to handle weird things about JSON in Vim pretty well. I'd give it a default that isn't a Dictionary and then use `if type(l:file) is v:t_dict`.